### PR TITLE
Filter ingress ports

### PR DESCRIPTION
**Pull request by ijsnellf**
Filter out non-HTTP ingress ports. This should fix the Envoy crashing observed in https://github.com/istio/issues/issues/20. Instead, Pilot will log a warning.

### CODE REVIEW

File: proxy/envoy/resolve.go [hunk 1]
```diff
@@ -10,13 +10,13 @@ import (
 	"github.com/golang/glog"
 )
 
-func resolveStatsdAddr(statsdAdrr string) (string, error) {
-	if statsdAdrr == "" {
+func resolveStatsdAddr(statsdAddr string) (string, error) {
```
Comment (ijsnellf) [inline proxy/envoy/resolve.go]: fixed typo

Comment (rshriram) [inline proxy/envoy/resolve.go]: ? Why remove the if block?

File: proxy/envoy/ingress.go [hunk 2]
```diff
@@ -276,6 +276,9 @@ func buildIngressRoute(ingress *proxyconfig.IngressRule,
 	if err != nil {
 		return nil, "", err
 	}
+	if servicePort.Protocol != model.ProtocolHTTP {
```
Comment (rshriram) [inline proxy/envoy/ingress.go]: Https?

File: proxy/envoy/resolve.go (hunk 1 above)
Comment (rshriram) [inline proxy/envoy/resolve.go]: Ignore

File: proxy/envoy/ingress.go (hunk 2 above)
Comment (kyessenov) [inline proxy/envoy/ingress.go]: Ingress is only for HTTP traffic in-cluster. It may expose HTTPS on the front side, but that's irrelevant here.

**Comment by kyessenov** [comment]
Disregard e2e failures, the test is broken.

**Pull request merged**